plugins/dell/fu-plugin-dell.c

```diff
@@ -566,9 +567,15 @@ fu_plugin_dell_detect_tpm (FuPlugin *plugin, GError **error)
 		return FALSE;
 	}
 
-	if (!data->smi_obj->fake_smbios)
-		system_id = (guint16) sysinfo_get_dell_system_id ();
-	else
+	system_id_str = fu_plugin_get_dmi_value (plugin,
+						 FU_HWIDS_KEY_PRODUCT_SKU);
+	if (!data->smi_obj->fake_smbios) {
+		if (system_id_str != NULL)
+			system_id = (guint16) g_ascii_strtoull (system_id_str,
```

Comment:
Wre we *sure* this always is an integer? I'd be happier if there was check for zero, or the endptr here.

Reply:
With how we use the value stored to this this internally I would be immensely surprised if this changed, but adding a check is still a good thing.  Added.